src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -220,28 +216,74 @@ public Claim saveClaim(
             .claimSubmissionOperationIndicators(ClaimSubmissionOperationIndicators.builder().build())
             .build();
 
-        Claim savedClaim = caseRepository.saveClaim(user, claim);
+        Claim savedClaim = caseRepository.saveClaim(user, claim, CaseEvent.CREATE_CASE);
+        createClaimEvent(authorisation, user, pinResponse.orElse(null), savedClaim);
+        trackClaimIssued(savedClaim.getReferenceNumber(), savedClaim.getClaimData().isClaimantRepresented());
+
+        return savedClaim;
+    }
+
+    @LogExecutionTime
+    @Transactional(transactionManager = "transactionManager")
+    public Claim saveLegalRepClaim(
+        String submitterId,
+        ClaimData claimData,
+        String authorisation
+    ) {
+        String externalId = claimData.getExternalId().toString();
+        User user = isExistingClaim(authorisation, externalId);
+
+        Optional<GeneratePinResponse> pinResponse = getPinResponse(claimData, authorisation);
+        Optional<String> letterHolderId = pinResponse.map(GeneratePinResponse::getUserId);
+        String submitterEmail = user.getUserDetails().getEmail();
+
+        Claim claim = Claim.builder()
+            .claimData(claimData)
+            .submitterId(submitterId)
+            .externalId(externalId)
+            .submitterEmail(submitterEmail)
+            .createdAt(nowInLocalZone())
+            .letterHolderId(letterHolderId.orElse(null))
+            .claimSubmissionOperationIndicators(ClaimSubmissionOperationIndicators.builder().build())
+            .build();
+
+        Claim savedClaim = caseRepository.saveClaim(user, claim, CaseEvent.CREATE_LEGAL_REP_CLAIM);
+        createClaimEvent(authorisation, user, pinResponse.orElse(null), savedClaim);
+        trackClaimIssued(savedClaim.getReferenceNumber(), savedClaim.getClaimData().isClaimantRepresented());
+
+        return savedClaim;
+    }
+
+    private User isExistingClaim(String authorisation, String externalId) {
+        User user = userService.getUser(authorisation);
+
+        caseRepository.getClaimByExternalId(externalId, user).ifPresent(claim -> {
+            throw new ConflictException(
+                String.format("Claim already exist with same external reference as %s", externalId));
+        });
+        return user;
+    }
+
+    private void createClaimEvent(String authorisation, User user, GeneratePinResponse pinResponse, Claim savedClaim) {
         ccdEventProducer.createCCDClaimIssuedEvent(savedClaim, user);
 
         if (asyncEventOperationEnabled) {
             eventProducer.createClaimCreatedEvent(
                 savedClaim,
-                pinResponse.map(GeneratePinResponse::getPin).orElse(null),
+//                Optional.ofNullable(pinResponse).map(GeneratePinResponse::getPin).orElse(null),
+                Optional.ofNullable(pinResponse).isPresent() ? pinResponse.getPin() : null,
```

Comment:
I need help here please - I don't know what is best to do here.... pinResponse can be null. I was trying line 273/274. Please let me know what is right approach.